### ISSUE TITLE
Always check for errors returned from epgsql_sock:send/3

### ISSUE THREAD

**Issue by dszoboszlay** [epgsql/epgsql#286]
`epgsql_sock` assumes that if a send operation fails and the connection gets closed, it would still receive an `ssl_error` or `tcp_error` message and therefore doesn't always check the return value of `send/3`.

However, erlang/otp#7431 demonstrated that this is not always the case, `ssl:send/2` can fail in a way that it only returns an `{error, Reason}` tuple and the connection gets closed without sending the `ssl_error` message to the controlling process. This leaves `epgsql_sock` in a broken state where it assumes the connection is still up, only idle, while in fact the connection is already closed.

Therefore all calls to `epgsql_sock:send/3` shall check the return value and stop the process.

**Comment by seriyps** [comment]
Yup, sounds reasonable. Just not sure if epgsql should fail gracefully or just crash...

Also, PR would be welcome!

**Comment by dszoboszlay** [comment]
I plan to implement a fix, but opened the issue first to serve as a TODO item for me. :+1:

**Comment by seriyps** [comment]
I think the easiest would be to do

```erlang
case do_send(...) of
  ok -> ok;
  {error, Reason} ->
    throw({stop, {error, Reason}, flush_queue(State#state{sock = undefined}, {error, sock_closed}) }
end
```

here https://github.com/epgsql/epgsql/blob/7ba52768cf0ea7d084df24d4275a88eef4db13c2/src/epgsql_sock.erl#L486 and above
